src/test/shell/bazel/bazel_coverage_cc_test_gcc.sh

```diff
@@ -110,6 +110,24 @@ was not found in actual coverage report:
 <$( cat "$output_file" )>"
 }
 
+# Asserts if the given expected coverage result is included in the given output
+# file, accounting for the fact that branch coverage is disabled for gcov v7
+#
+# - expected coverage The expected result that must be included in the output.
+# - output_file       The location of the coverage output file.
+function assert_cc_coverage_result() {
+  local expected_coverage="${1}"; shift
+  local output_file="${1}"; shift
+
+  # we disable branch coverage when using gcov 7 so we should strip all branch
+  # information from the "expected" result before checking.
```

Comment:
nit: please include an example `gcov -v` output in the comment here and explain that you are extracting the major version

Reply:
same in all places below

Reply:
Done.